### DISCUSSION TITLE
Rendu côté serveur de toutes les pages

### DISCUSSION THREAD

**Discussion by mquandalle** [betagouv/mon-entreprise#2720] [Idées]
Actuellement seule une liste de page prédéfinies bénéficie du rendu côté serveur, il faudrait le supporter sur l'ensemble des pages.

Possibilités :
- Utiliser une lambda pour faire le rendu (avec `Cache-control: immutable` pour que le rendu de chaque page ne soit pas qu'une seule fois par déploiement)
- Génération lors du déploiement, mais il faut vérifier le temps que ça prend

**Comment by johangirod** [comment]
À première vue, je serai partisan de la solution prérendu de toutes les pages listées dans `sitePaths` au build. De cette manière, on pourrait générer automatiquement le `netlify.toml` qui fait les redirections. On ne ferait en revanche pas de prérendu pour les pages de doc.

**Comment by mquandalle** [comment]
J'aimerai bien enlever la liste des redirections dans le `netlify.toml`. Ça me semble possible en sortant une arborescence qui a la bonne structure :

```
mon-entreprise/
  netlify.toml (sans les redirections "prerender/")
  créer/
    index.html
    auto-entreprise/
      index.html
  simulateurs/
    salarié/
      index.html
infrance/
  create/
    index.html
```

Il faut un peut remanipuler la sortie `vitejs` pour obtenir ça, mais je pense que ça se fait assez bien.

**Comment by johangirod** [comment]
Excellente idée ! On peut utiliser les `sitePaths` pour générer cette arborescence.
